src/util/Util.cpp

```diff
@@ -37,34 +37,37 @@ static bool execInUiThreadCallback(CallbackUiData* cb)
  *
  * Make sure the container class is not deleted before the UI stuff is finished!
  */
-void Util::execInUiThread(std::function<void()> callback)
+void Util::execInUiThread(std::function<void()>&& callback)
 {
-	gdk_threads_add_idle((GSourceFunc) execInUiThreadCallback, new CallbackUiData(callback));
+	gdk_threads_add_idle((GSourceFunc) execInUiThreadCallback, new CallbackUiData(std::move(callback)));
 }
 
-void Util::cairo_set_source_rgbi(cairo_t* cr, int c)
+void Util::cairo_set_source_rgbi(cairo_t* cr, int color)
 {
-	double r = ((c >> 16) & 0xff) / 255.0;
-	double g = ((c >> 8) & 0xff) / 255.0;
-	double b = (c & 0xff) / 255.0;
+	auto tmp = static_cast<unsigned>(color);
+
+	double r = ((tmp >> 16u) & 0xffu) / 255.0;
+	double g = ((tmp >> 8u) & 0xffu) / 255.0;
+	double b = (tmp & 0xffu) / 255.0;
 
 	cairo_set_source_rgb(cr, r, g, b);
 }
 
 
 void Util::apply_rgb_togdkrgba(GdkRGBA& col, int color)
 {
-	col.red = ((color >> 16) & 0xFF) / 255.0;
-	col.green = ((color >> 8) & 0xFF) / 255.0;
-	col.blue = (color & 0xFF) / 255.0;
+	auto tmp = static_cast<unsigned>(color);
```

Comment:
This casts are not needed, it makes the code more complicated, and therefore worse to read. Don't do this, except if you have a really good explanation.

Reply:
Shifting an unsigned behaves like you think the int behaves. But to shift an int behaves differently.
Shifting an int 1000 0000 1110 0000 to the right by one:
1000 0000 0111 0000 
Shifting an unsigned: 1000 0000 1110 0000 :
0100 0000 0111 0000
So that's why no one should shift signed values if he really don't know what he's doing.
And yellow marked code is very annoying.

Reply:
No, this is wrong.

´´´
#include <stdio.h>

int main() {
	unsigned a = 0b1000000011100000;
	int b = 0b1000000011100000;

	printf("a1=%04x\n", a);
	printf("b1=%04x\n", b);
	
	a = a >> 1;
	b = b >> 1;
	
	printf("a2=%04x\n", a);
	printf("b2=%04x\n", b);
	
	return 0;
}
´´´
g++ main.cpp -o main
./main

´´´
a1=80e0
b1=80e0
a2=4070
b2=4070
´´´

Reply:
Oh you misunderstood me, i used a 16bit int so i dont need to write so much 0's.
Look at this site: https://www.godbolt.org/z/fyP17H there is the proof for you.

Reply:
```
int square(int num) {
    int i = 0xf000000f;
    return i>>31u;
} // returns -1

int square2(int num) {
    unsigned i = 0xf000000f;
    return i>>31u;
} //returns 1```